### PR TITLE
Olympia add `WorkingGroupOpeningMetadata.title`

### PR DESCRIPTION
**Pull request by thesan**
See Joystream/pioneer#2395

### CODE REVIEW

**Review by ondratra — APPROVED** [review]
LGTM

File: query-node/mappings/src/workingGroups.ts [hunk 1]
```diff
@@ -175,13 +175,15 @@ export async function createWorkingGroupOpeningMetadata(
...
     shortDescription,
+    title,
   } = metadata
 
   const openingMetadata = new WorkingGroupOpeningMetadata({
     createdAt: eventTime,
     updatedAt: eventTime,
     originallyValid,
     applicationDetails: applicationDetails || undefined,
+    title: title ?? undefined,
```
Comment (Lezek123) [inline query-node/mappings/src/workingGroups.ts]: This is a minor inconsistency with other fields (like `description` etc.), where empty string (`''`) is treated as _no-value_ (`null`)

**Review by Lezek123 — CHANGES_REQUESTED** [review]
I know it's not within the scope of this PR, but I noticed that working group opening metadata processing is not secured against `/u0000` character.

Could you perhaps make it part of `deserializeMetadata` to iterate over all metadata fields and make sure that in case `typeof field === 'string'` it's beeing santizied with `prepareString`?

File: query-node/mappings/src/common.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { WorkingGroup, WorkerId, ThreadId } from '@joystream/types/augment/all'
 import { Worker, Event, Network } from 'query-node/dist/model'
 import { BaseModel } from '@joystream/warthog'
 import { metaToObject } from '@joystream/metadata-protobuf/utils'
-import { AnyMetadataClass, DecodedMetadataObject } from '@joystream/metadata-protobuf/types'
+import { AnyMessage, AnyMetadataClass, DecodedMetadataObject } from '@joystream/metadata-protobuf/types'
```
Comment (Lezek123) [inline query-node/mappings/src/common.ts]: Unused import
```suggestion
import { AnyMetadataClass, DecodedMetadataObject } from '@joystream/metadata-protobuf/types'
```

File: query-node/mappings/src/common.ts [hunk 3]
```diff
@@ -198,7 +198,13 @@ export function deserializeMetadata<T>(
   metadataBytes: Bytes
 ): DecodedMetadataObject<T> | null {
   try {
-    return metaToObject(metadataType, metadataType.decode(metadataBytes.toU8a(true)))
+    const message = metadataType.decode(metadataBytes.toU8a(true))
+    Object.keys(message).forEach((key) => {
+      if (message.hasOwnProperty(key) && typeof message[key] === 'string') {
```
Comment (Lezek123) [inline query-node/mappings/src/common.ts]: Linter error
```suggestion
      if (key in message && typeof message[key] === 'string') {
```

File: query-node/mappings/src/common.ts [hunk 4]
```diff
@@ -198,7 +198,13 @@ export function deserializeMetadata<T>(
   metadataBytes: Bytes
 ): DecodedMetadataObject<T> | null {
   try {
-    return metaToObject(metadataType, metadataType.decode(metadataBytes.toU8a(true)))
+    const message = metadataType.decode(metadataBytes.toU8a(true))
+    Object.keys(message).forEach((key) => {
+      if (typeof message[key] === 'string') {
+        message[key] = bytesToString(message[key])
```
Comment (Lezek123) [inline query-node/mappings/src/common.ts]:
```suggestion
        message[key] = prepareString(message[key])
```

**Review by Lezek123 — CHANGES_REQUESTED** [review]

**Comment by thesan** [comment]
@Lezek123 sorry about that commit last night, I was pretty tired :sweat_smile:

**Review by Lezek123 — APPROVED** [review]
LGTM!

**Pull request merged**